components/ai_rewriter/renderer/ai_rewriter_agent.cc

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/ai_rewriter/renderer/ai_rewriter_agent.h"
+
+#include <utility>
+
+#include "base/functional/bind.h"
+#include "content/public/browser/content_browser_client.h"
+#include "content/public/renderer/render_frame.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "third_party/blink/public/web/web_document.h"
+#include "third_party/blink/public/web/web_element.h"
+#include "third_party/blink/public/web/web_frame.h"
+#include "third_party/blink/public/web/web_frame_widget.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "ui/gfx/geometry/rect_f.h"
+
+namespace ai_rewriter {
+namespace {
+gfx::RectF GetBounds(content::RenderFrame* frame) {
+  auto document = frame->GetWebFrame()->GetDocument();
+
+  auto focused = document.FocusedElement();
+  if (focused.IsNull()) {
+    return gfx::RectF();
+  }
+
+  return frame->ElementBoundsInWindow(focused);
+}
+}  // namespace
+
+AIRewriterAgent::AIRewriterAgent(content::RenderFrame* render_frame,
+                                 service_manager::BinderRegistry* registry)
+    : render_frame_(render_frame) {
```

Comment:
this should be `content::RenderFrameObserver(render_frame)` and you should call `delete this` in `OnDestruct()` override.